homeassistant/components/nexia/switch.py

```diff
@@ -25,16 +28,31 @@ async def async_setup_entry(
     """Set up switches for a Nexia device."""
     coordinator = config_entry.runtime_data
     nexia_home = coordinator.nexia_home
-    entities: list[NexiaHoldSwitch | NexiaEmergencyHeatSwitch] = []
+    entities: list[SwitchEntity] = []
+    room_iq_zones: dict[int, NexiaRoomIQHarmonizer] = {}
     for thermostat_id in nexia_home.get_thermostat_ids():
         thermostat: NexiaThermostat = nexia_home.get_thermostat_by_id(thermostat_id)
         if thermostat.has_emergency_heat():
             entities.append(NexiaEmergencyHeatSwitch(coordinator, thermostat))
         for zone_id in thermostat.get_zone_ids():
             zone: NexiaThermostatZone = thermostat.get_zone_by_id(zone_id)
             entities.append(NexiaHoldSwitch(coordinator, zone))
+            if len(zone_sensors := zone.get_sensors()) > 1:
+                entities.extend(
+                    NexiaRoomIQSwitch(coordinator, zone, sensor, room_iq_zones)
+                    for sensor in zone_sensors
+                )
 
     async_add_entities(entities)
+    for harmonizer in room_iq_zones.values():
+
+        async def _stop_obj(_: Event, obj: NexiaRoomIQHarmonizer = harmonizer) -> None:
+            """Run the shutdown method when preparing to stop."""
+            await obj.async_shutdown()
+
+        config_entry.async_on_unload(
+            hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, _stop_obj)
+        )
```

Comment:
Let's only install one of these and have it shutdown them all.

Also if the integration gets unloaded after the stop event the async_on_unload call will fail because the listener will already be cancelled so we should use async_listen.